Debouncer/DebouncerImpl.h

```diff
@@ -69,21 +70,29 @@ class Debouncer
     , mode(mode)
     {}
 
-    bool read() { return stable_state; }
+    bool read() const { return stable_state; }
 
-    bool edge() { return is_stable_edge; }
-    bool rising() { return (stable_state == HIGH) && is_stable_edge; }
-    bool falling() { return (stable_state == LOW) && is_stable_edge; }
+    bool edge() const { return is_stable_edge; }
+    bool rising() const { return (stable_state == HIGH) && is_stable_edge; }
+    bool falling() const { return (stable_state == LOW) && is_stable_edge; }
 
     void update()
     {
         bool curr_state = digitalRead(pin_target);
```

Comment:
```suggestion
        bool curr_state = digitalRead(pin_target);
        uint32_t curr_ms = millis();
```
And I also think these lines will be better just to init the variables (no `digitalRead()`).

Reply:
Good idea.

Reply:
Regarding #ifdef SOME_BOARDS_OR_ARCH, do you know which boards do and do not support ATOMIC_BLOCK?

I've also asked here:
https://github.com/arduino/reference-en/issues/761